src/Riok.Mapperly/Descriptors/WellKnownTypes.cs

```diff
@@ -17,6 +18,16 @@ internal WellKnownTypes(Compilation compilation)
 
     public INamedTypeSymbol? TimeOnly => TryGet("System.TimeOnly");
 
+    public IArrayTypeSymbol GetArrayType(ITypeSymbol type) =>
+        _compilation.CreateArrayTypeSymbol(type, elementNullableAnnotation: type.NullableAnnotation);
+
+    public bool IsSpanType(ITypeSymbol symbol) =>
+        SymbolEqualityComparer.Default.Equals(symbol.OriginalDefinition, Get(typeof(Span<>)))
+        || SymbolEqualityComparer.Default.Equals(symbol.OriginalDefinition, Get(typeof(ReadOnlySpan<>)));
+
+    public ITypeSymbol? GetEnumeratedType(ITypeSymbol type) =>
```

Comment:
This could and probably should be used here too:
* `Riok.Mapperly.Descriptors.MappingBuilders.DictionaryMappingBuilder.GetEnumerableKeyValueTypes`
* `Riok.Mapperly.Descriptors.Enumerables.EnsureCapacity.EnsureCapacityBuilder.TryBuildEnsureCapacity`